featuretools/entityset/entityset.py

```diff
@@ -1203,14 +1192,67 @@ def update_dataframe(self, entity_id, df, already_sorted=False, recalculate_last
         # Make sure column ordering matches variable ordering
         self[entity_id].df = df[[v.id for v in variables]]
         self[entity_id].set_index(self[entity_id].index)
-        if self[entity_id].time_index is not None:
-            self[entity_id].set_time_index(self[entity_id].time_index, already_sorted=already_sorted)
-
+        self[entity_id]._already_sorted = already_sorted
         self.set_secondary_time_index(self[entity_id], self[entity_id].secondary_time_index)
+        self._check_time_index()
+
         if recalculate_last_time_indexes and self[entity_id].last_time_index is not None:
             self.add_last_time_indexes(updated_entities=[self[entity_id].id])
         self.reset_data_description()
 
+    def _get_time_type(self, entity, variable_id):
+        if not isinstance(entity.df, pd.DataFrame) or entity.df.empty:
+            dtype = entity[variable_id]._default_pandas_dtype
+            time_to_check = vtypes.DEFAULT_DTYPE_VALUES[dtype]
+        else:
+            time_to_check = entity.df[variable_id].iloc[0]
+
+        time_type = _check_time_type(time_to_check)
+        if time_type is None:
+            info = "%s time index not recognized as numeric or datetime"
+            raise TypeError(info % entity.id)
+        return time_type
+
+    def _check_time_index(self):
+        for entity in self.entity_dict.values():
```

Comment:
Yes, we can add a method to check time indexes for a single entity. It can work as a private method on either the Entity or EntitySet. We will still need to separate the logic of checking whether the time index is the same across entities. This seems like something that can happen in `_check_time_indexes`